tests/parser-tests.lisp

```diff
@@ -1,21 +1,35 @@
 (in-package #:coalton-tests)
 
 (deftest test-parser ()
-  (let* ((glob (merge-pathnames "tests/parser/*.bad.coalton" (asdf:system-source-directory "coalton/tests")))
+  (labels ((test-files (pattern)
+             (directory (merge-pathnames pattern (asdf:system-source-directory "coalton/tests"))))
 
-         (files (directory glob)))
+           (parse-file (file)
+             (with-open-file (stream file)
+               (parser:with-reader-context stream
+                 (parser:read-program stream (error:make-coalton-file :stream stream :name (namestring file)) :mode :file))))
 
-    (loop :for file :in files
-          :do (with-open-file (stream file)
-                (signals parser:parse-error
-                  (parser:with-reader-context stream
-                    (parser:read-program stream (error:make-coalton-file :stream stream :name (namestring file)) :mode :file))))))
+           (error-string (condition)
```

Comment:
I think this can just be `(format nil "~A" condition)`

Reply:
princ-to-string

Reply:
done